main.go

```diff
@@ -103,8 +99,9 @@ func main() {
 	sig := <-sigChan
 	waitSeconds := config.RelayServerConfig.GracefulShutdownWaitSeconds
 	log.Printf("Sig %v received, shutting down, graceful shutdown wait: %v seconds\n", sig, waitSeconds)
-	select {
-	case <-time.After(time.Duration(waitSeconds) * time.Second):
-	}
+
 	relayServer.Shutdown()
+
+	<-time.After(time.Duration(waitSeconds) * time.Second)
```

Comment:
Sorry, you're right, `server.Shutdown()` will block indefinitely if connections are active. I have changed it back.

```
// Shutdown gracefully shuts down the server without interrupting any
// active connections. Shutdown works by first closing all open
// listeners, then closing all idle connections, and then waiting
// indefinitely for connections to return to idle and then shut down.
// If the provided context expires before the shutdown is complete,
// Shutdown returns the context's error, otherwise it returns any
// error returned from closing the Server's underlying Listener(s).
```